pkg/status/combinedstatus-resolution.go

```diff
@@ -346,10 +362,10 @@ func (c *combinedStatusResolution) requiresSourceObjectMetaOrSpec() bool {
 	defer c.RUnlock()
 
 	pred := func(s string) bool {
-		return strings.Contains(s, fmt.Sprintf("%s.metadata", expressionRootName)) ||
-			strings.Contains(s, fmt.Sprintf("%s.spec", expressionRootName)) ||
-			strings.Contains(s, fmt.Sprintf("%s.kind", expressionRootName)) ||
-			strings.Contains(s, fmt.Sprintf("%s.apiVersion", expressionRootName))
+		return strings.Contains(s, fmt.Sprintf("%s.metadata", sourceObjectKey)) ||
+			strings.Contains(s, fmt.Sprintf("%s.spec", sourceObjectKey)) ||
+			strings.Contains(s, fmt.Sprintf("%s.kind", sourceObjectKey)) ||
+			strings.Contains(s, fmt.Sprintf("%s.apiVersion", sourceObjectKey))
```

Comment:
Writing `sourceObjectKey + ".metadata"` makes a constant expression, which does not require CPU and heap activity every time requiresSourceObjectMetaOrSpec is invoked.

Reply:
This function specifically checks for spec and metadata. It should later be rewritten to check if any augmentation is used and make that process more efficient.

Reply:
@MikeSpreitzer  Fixed.